jsonnet/thanos/config.libsonnet

```diff
@@ -0,0 +1,277 @@
+local service(name, namespace, labels, selector, ports) = {
+  apiVersion: 'v1',
+  kind: 'Service',
+  metadata: {
+    name: name,
+    namespace: namespace,
+    labels: labels,
+  },
+  spec: {
+    ports+: ports,
+    selector: selector,
+  },
+};
+
+{
+	_config+:: {
+		namespace: 'default',
+		prometheusName: 'prometheus-self',
+		thanosQueryName: 'thanos-query',
+		thanosRulerName: 'thanos-ruler',
+		thanosSidecarName: 'thanos-sidecar',
+		versions+:: {
+			thanos: 'v0.11.2',
+		},
+
+		imageRepos+:: {
+			thanos: 'quay.io/thanos/thanos',
+		},
+		labels+:: {
+            prometheusLabels: {
+                'prometheus': 'self',
+            },
+            commonLabels:{
+             'prometheus': 'self',
+             'app': 'prometheus'
+            },
+            queryLabels: {'app':'thanos-query'},
+            sidecarLabels: {'app':'thanos-sidecar'},
+            rulerLabels: {'app':'thanos-ruler'},
+        },
+
+	},
+	thanos+:: {
+		local po = self,
+		namespace:: $._config.namespace,
+		image:: $._config.imageRepos.thanos,
+		version:: $._config.versions.thanos,
+	    commonLabels:: $._config.labels.commonLabels,
+	    prometheusLabels:: $._config.labels.prometheusLabels,
+	    queryLabels:: $._config.labels.queryLabels,
+	    sidecarLabels:: $._config.labels.sidecarLabels,
+	    rulerLabels:: $._config.labels.rulerLabels,
+	    prometheusName:: $._config.prometheusName,
+	    thanosQueryName:: $._config.thanosQueryName,
+	    thanosRulerName:: $._config.thanosRulerName,
+	    thanosSidecarName:: $._config.thanosSidecarName,
+		prometheus+:: {
```

Comment:
nit: seems like formatting is off, running `jsonnetfmt -i jsonnet/thanos/config.libsonnet` should fix that.

Reply:
ok